lib/block_view_lockups.go

```diff
@@ -0,0 +1,2193 @@
+	ProfilePublicKey               *PublicKey
+	UnlockTimestampNanoSecs        int64
+	LockedCoinsToTransferBaseUnits *uint256.Int
+}
+
+func (txnData *CoinLockupTransferMetadata) GetTxnType() TxnType {
+	return TxnTypeCoinLockupTransfer
+}
+
+func (txnData *CoinLockupTransferMetadata) ToBytes(preSignature bool) ([]byte, error) {
+	var data []byte
+	data = append(data, EncodeByteArray(txnData.RecipientPublicKey.ToBytes())...)
+	data = append(data, EncodeByteArray(txnData.ProfilePublicKey.ToBytes())...)
+	data = append(data, IntToBuf(txnData.UnlockTimestampNanoSecs)...)
+	data = append(data, VariableEncodeUint256(txnData.LockedCoinsToTransferBaseUnits)...)
+	return data, nil
+}
+
+func (txnData *CoinLockupTransferMetadata) FromBytes(data []byte) error {
+	rr := bytes.NewReader(data)
+
+	// RecipientPublicKey
+	recipientPublicKeyBytes, err := DecodeByteArray(rr)
+	if err != nil {
+		return errors.Wrap(err, "CoinLockupTransferMetadata.FromBytes: Problem reading RecipientPublicKey")
+	}
+	txnData.RecipientPublicKey = NewPublicKey(recipientPublicKeyBytes)
+
+	// ProfilePublicKey
+	profilePublicKeyBytes, err := DecodeByteArray(rr)
+	if err != nil {
+		return errors.Wrap(err, "CoinLockupTransferMetadata.FromBytes: Problem reading ProfilePublicKey")
+	}
+	txnData.ProfilePublicKey = NewPublicKey(profilePublicKeyBytes)
+
+	// UnlockTimestampNanoSecs
+	txnData.UnlockTimestampNanoSecs, err = ReadVarint(rr)
+	if err != nil {
+		return errors.Wrap(err, "CoinLockupTransferMetadata.FromBytes: Problem reading UnlockTimestampNanoSecs")
+	}
+
+	// LockedDAOCoinToTransferBaseUnits
+	txnData.LockedCoinsToTransferBaseUnits, err = VariableDecodeUint256(rr)
+	if err != nil {
+		return errors.Wrap(err, "CoinLockupTransferMetadata.FromBytes: Problem reading LockedDAOCoinToTransferBaseUnits")
+	}
+
+	return nil
+}
+
+func (txnData *CoinLockupTransferMetadata) New() DeSoTxnMetadata {
+	return &CoinLockupTransferMetadata{}
+}
+
+//
+// TYPES: CoinUnlockMetadata
+//
+
+type CoinUnlockMetadata struct {
+	ProfilePublicKey *PublicKey
+}
+
+func (txnData *CoinUnlockMetadata) GetTxnType() TxnType {
+	return TxnTypeCoinUnlock
+}
+
+func (txnData *CoinUnlockMetadata) ToBytes(preSignature bool) ([]byte, error) {
+	var data []byte
+	data = append(data, EncodeByteArray(txnData.ProfilePublicKey.ToBytes())...)
+	return data, nil
+}
+
+func (txnData *CoinUnlockMetadata) FromBytes(data []byte) error {
+	rr := bytes.NewReader(data)
+
+	// ProfilePublicKey
+	profilePublicKeyBytes, err := DecodeByteArray(rr)
+	if err != nil {
+		return errors.Wrap(err, "CoinUnlockMetadata.FromBytes: Problem reading ProfilePublicKey")
+	}
+	txnData.ProfilePublicKey = NewPublicKey(profilePublicKeyBytes)
+
+	return nil
+}
+
+func (txnData *CoinUnlockMetadata) New() DeSoTxnMetadata {
+	return &CoinUnlockMetadata{}
+}
+
+//
+// CoinLockup Transaction Logic
+//
+
+func (bav *UtxoView) _connectCoinLockup(
+	txn *MsgDeSoTxn, txHash *BlockHash,
+	blockHeight uint32, blockTimestamp int64,
```

Comment:
Always include the units of measurement in variable names like this
```suggestion
	blockHeight uint32, blockTimestampNanoSecs int64,
```